toolkit/tools/internal/ccachemanager/ccachemanager.go

```diff
@@ -0,0 +1,769 @@
+		}
+
+		ccachePkgGroup.UpdateTarPaths(m.Configuration.RemoteStoreConfig, m.LocalDownloadsDir, m.LocalUploadsDir)
+	}
+
+	m.CurrentPkgGroup = ccachePkgGroup
+
+	return nil
+}
+
+func loadConfiguration(configFileName string) (configuration *CCacheConfiguration, err error) {
+
+	logger.Log.Infof("  loading ccache configuration file: %s", configFileName)
+
+	err = jsonutils.ReadJSONFile(configFileName, &configuration)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to load file:\n%w", err)
+	}
+
+	logger.Log.Infof("    Type           : %s", configuration.RemoteStoreConfig.Type)
+	logger.Log.Infof("    TenantId       : %s", configuration.RemoteStoreConfig.TenantId)
+	logger.Log.Infof("    UserName       : %s", configuration.RemoteStoreConfig.UserName)
+	logger.Log.Infof("    StorageAccount : %s", configuration.RemoteStoreConfig.StorageAccount)
+	logger.Log.Infof("    ContainerName  : %s", configuration.RemoteStoreConfig.ContainerName)
+	logger.Log.Infof("    Tagsfolder     : %s", configuration.RemoteStoreConfig.TagsFolder)
+	logger.Log.Infof("    DownloadEnabled: %v", configuration.RemoteStoreConfig.DownloadEnabled)
+	logger.Log.Infof("    DownloadLatest : %v", configuration.RemoteStoreConfig.DownloadLatest)
+	logger.Log.Infof("    DownloadFolder : %s", configuration.RemoteStoreConfig.DownloadFolder)
+	logger.Log.Infof("    UploadEnabled  : %v", configuration.RemoteStoreConfig.UploadEnabled)
+	logger.Log.Infof("    UploadFolder   : %s", configuration.RemoteStoreConfig.UploadFolder)
+	logger.Log.Infof("    UpdateLatest   : %v", configuration.RemoteStoreConfig.UpdateLatest)
+	logger.Log.Infof("    KeepLatestOnly : %v", configuration.RemoteStoreConfig.KeepLatestOnly)
+
+	return configuration, err
+}
+
+func compressDir(sourceDir string, archiveName string) (err error) {
+
+	// Ensure the output file does not exist...
+	_, err = os.Stat(archiveName)
+	if err == nil {
+		err = os.Remove(archiveName)
+		if err != nil {
+			return fmt.Errorf("Unable to delete ccache out tar:\n%w", err)
+		}
+	}
+
+	// Create the archive...
+	logger.Log.Infof("  compressing (%s) into (%s).", sourceDir, archiveName)
+	compressStartTime := time.Now()
+	tarArgs := []string{
+		"cf",
+		archiveName,
+		"-C",
+		sourceDir,
+		"."}
+
+	_, stderr, err := shell.Execute("tar", tarArgs...)
+	if err != nil {
+		return fmt.Errorf("Unable compress ccache files itno archive:\n%s", stderr)
+	}
+	compressEndTime := time.Now()
+	logger.Log.Infof("  compress time: %s", compressEndTime.Sub(compressStartTime))
+	return nil
+}
+
+func uncompressFile(archiveName string, targetDir string) (err error) {
+	logger.Log.Infof("  uncompressing (%s) into (%s).", archiveName, targetDir)
+	uncompressStartTime := time.Now()
+	tarArgs := []string{
+		"xf",
+		archiveName,
+		"-C",
+		targetDir,
+		"."}
+
+	_, stderr, err := shell.Execute("tar", tarArgs...)
+	if err != nil {
+		return fmt.Errorf("Unable extract ccache files from archive:\n%s", stderr)
```

Comment:
print error with `%w` for consistency?

Reply:
Herre's the error I'm seeing `fmt.Errorf format %w has arg stderr of wrong type string`.